common/chunkStatusLogger.go

```diff
@@ -337,13 +347,19 @@ func (csl *chunkStatusLogger) main(chunkLogPath string) {
 			alwaysFlushFromNowOn = true
 			doFlush()
 			csl.flushDone <- struct{}{}
-			continue // TODO can become break (or be moved to later if we close unsaved entries, once we figure out how we got stuff written to us after CloseLog was called)
+			select {
+			case <-csl.done:
+				return
```

Comment:
I think we don't need to have a signalling channel "done".
When CloseLogger() closes the channel, the for loop will exit, so we will infact never go here and check for the done channel.